es-mapping-gen.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+
+import os
+import json
+import yaml
+import requests
+from shutil import copyfile
+
+
+def expand_es_mapping(mapping, source_models, mapper):
+    next_mappings = {}
+    config = {}
+    for k, v in mapping.iteritems():
+        if k == 'properties':
+            next_mappings = v
+        else:
+            config[k] = v
+
+    for next_mapping in next_mappings:  # recursion
+        expand_es_mapping(next_mappings[next_mapping], source_models, mapper)
+
+    # add new properties
+    if config.get('source'):
+        source_model = source_models.get(config.get('source'), {}).get('properties', {})
+        exclude_list = config.get('exclude')
+
+        for p in source_model:
+            if p in exclude_list:
+                continue
+            source_property = source_model.get(p)
+            source_type = source_property.get('type')
+            source_format = 'format:%s' % source_property.get('format') \
+                if source_property.get('format') else 'format:null'
+
+            es_type = mapper.get('field_type_map').get(source_type, {}).get(source_format)
+            if es_type and p not in next_mappings:  # do NOT overwrite manually defined existing mapping type
+                next_mappings.update({p: es_type})
+
+        mapping.pop('source', None)
+        mapping.pop('exclude', None)
+
+
+class SchemaGenerator(object):
+    def __init__(self, mapper_file):
+        self._read_mapper(mapper_file)
+        self._source_models = {}
+        self._target_models = {}
+        self._get_sources()
+        #print json.dumps(self._source_models, indent=2)
+
+    @property
```

Comment:
What's the advantage of having private attributes if the are transparently translated directly to public ones?

Reply:
not quite matter here given the simplicity of the whole thing. Just the way I usually do for `getter`, `setter` in Python: https://stackoverflow.com/questions/2627002/whats-the-pythonic-way-to-use-getters-and-setters